controllers/authContoller.js

```diff
@@ -0,0 +1,27 @@
+const jwt =require('jsonwebtoken');
+const Joi = require('joi');
+const bcrypt = require('bcrypt');
+const {User, } = require('../models/user');
+
+exports.createUser = async (req, res) => {
+    const { error } = validate(req.body); 
+    if (error) return res.status(400).send(error.details[0].message);
+    let user = await User.findOne({name: req.body.name});
+  
+    if (!user) return res.status(400).send('invalid name or password.');
+    
+    const validPsaaword = await bcrypt.compare(req.body.password, user.password);
+   if (!validPsaaword) return res.status(400).send('invalid name or password.');
+     
+   token = jwt.sign({_id: user.id}, 'estateCitadel')
+    res.send(token);
+  };
+ 
+
+  function validate(req) {
```

Comment:
this should be in a separate file. You could create a folder called `validators` and add the validators for all schemas there.